docs/design/kotlin-smithy-sdk.md

```diff
@@ -97,39 +97,30 @@ structure MyStruct {
 Given the above Smithy structure shapes above we would generate the following:
 
 ```kotlin
-class Baz private constructor(builder: BuilderImpl) {
+class Baz private constructor(builder: Builder) {
     val quux: String? = builder.quux
 
     override fun toString(): String = "Baz(quux=$quux)"
 
     companion object {
-        @JvmStatic
-        fun builder(): Builder = BuilderImpl()
-
-        operator fun invoke(block: DslBuilder.() -> Unit): Baz = BuilderImpl().apply(block).build()
+        operator fun invoke(block: Builder.() -> Unit): Baz = Builder().apply(block).build()
```

Comment:
if `Builder` is now public this should probably be `inline operator fun` (only commenting here but obviously would apply everywhere)